### PR TITLE
Hw02

### PR DESCRIPTION
**Pull request by MateAcademy**

### CODE REVIEW

File: src/main/java/mate/academy/hw02/ImmutableInt.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package mate.academy.hw02;
+
+public class ImmutableInt {
+    private final int value;
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/ImmutableInt.java:4]: Add ability to change one of the fields, so the class stays immutable

File: src/main/java/mate/academy/hw02/Popcorn.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package mate.academy.hw02;
+
+public class Popcorn {
+    void doSomthing() {
+        System.out.println("popcorn");
+    }
+    void secondMethod() {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/Popcorn.java]: Popcorn  secondMethod, are you sure this is valid behavior for popcorn ?

File: src/main/java/mate/academy/hw02/Popcorn.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            void secondMethod() {
+                System.out.println("first");
+            }
+        };
+        popcorn.doSomthing();
+        popcorn.secondMethod();
+    }
+
+
+    private void doThomthimg() {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/Popcorn.java]: typo, use real method name

File: src/main/java/mate/academy/hw02/nestedClass/Electrocar.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package mate.academy.hw02.nestedClass;
+
+public class Electrocar {
+    private int id;
+    private static int test;
+
+    //вложенный нестатический класс
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/nestedClass/Electrocar.java]: Only English

File: src/main/java/mate/academy/hw02/nestedClass/Electrocar.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+    public Electrocar(int id) {
+        this.id = id;
+    }
+    public void start() {
+        Motor motor = new Motor();
+        motor.startMotor();
+        final int power = 1;
+
+        class SomeClass {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/nestedClass/Electrocar.java]: Use normal class names please

File: src/main/java/mate/academy/hw02/nestedClass/Electrocar.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                System.out.println(power);
+                System.out.println(id);
+            }
+        }
+        SomeClass someObject = new SomeClass();
+        test(someObject);
+        System.out.println("Electrocar " + id + " is starting...");
+    }
+
+    private void test(Object object) {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/nestedClass/Electrocar.java:42]: what it does ?

File: src/main/java/mate/academy/hw02/patternBuilder/Main.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+
+    public String getView() {
+        return view;
+    }
+
+    public String getAspect() {
+        return aspect;
+    }
+
+    private SportCar(Builder builder) {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/patternBuilder/Main.java:61]: Create SportCar in builder not in SportCar constructor

File: src/main/java/mate/academy/hw02/sort/BuubleSort.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        System.out.println("массив после пузырьковой сортировки : " + Arrays.toString(array));
+
+        System.out.println("Время выполнения программы:  " + (System.nanoTime() - t1) / 1_000_000);
+    }
+
+    // метод пузырьковой сортировки
+    public static void bubbleSort(int[] num) {
+
+        int j;
+        boolean flag = true;   // устанавливаем наш флаг в true для первого прохода по массиву
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/sort/BuubleSort.java]: remove comments,code should be self descriptive

File: src/main/java/mate/academy/hw02/sort/SliyaniemSort.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        if (arr.length < 2) return arr;
+        int m = arr.length / 2;
+        int[] arr1 = Arrays.copyOfRange(arr, 0, m);
+        int[] arr2 = Arrays.copyOfRange(arr, m, arr.length);
+        return merge(sort(arr1), sort(arr2));
+    }
+
+    //слияние двух массивов в один отсортированный
+    public static int[] merge(int[] arr1, int arr2[]) {
+        int n = arr1.length + arr2.length;
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/sort/SliyaniemSort.java]: use descriptive variables names

File: src/main/java/mate/academy/hw02/sort/SliyaniemSort.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
+package mate.academy.hw02.sort;
+
+import java.util.Arrays;
+
+public class SliyaniemSort {
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/sort/SliyaniemSort.java]: Not a valid name

**Review by Siril-O — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/hw02/ImmutableInt.java (hunk 1 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/ImmutableInt.java:4]: I schanged https://github.com/MateAcademy/lessons/pull/15/commits/53b96b74baa7788d478db63a5e5f302331d839bb

File: src/main/java/mate/academy/hw02/Popcorn.java (hunk 2 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/Popcorn.java]: I've changed https://github.com/MateAcademy/lessons/pull/15/commits/4dda5d27c75b8e58daa48b9ed20bb1d29396df18

File: src/main/java/mate/academy/hw02/Popcorn.java (hunk 3 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/Popcorn.java]: this is example local class

File: src/main/java/mate/academy/hw02/nestedClass/Electrocar.java (hunk 4 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/nestedClass/Electrocar.java]: https://github.com/MateAcademy/lessons/pull/15/commits/09f6f64902f1bd6db8c6dab6b7caee12f71d582b

File: src/main/java/mate/academy/hw02/nestedClass/Electrocar.java (hunk 5 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/nestedClass/Electrocar.java]: ok,  I've changed

File: src/main/java/mate/academy/hw02/nestedClass/Electrocar.java (hunk 6 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/nestedClass/Electrocar.java:42]: I deleted this method it was a mistake

File: src/main/java/mate/academy/hw02/sort/SliyaniemSort.java (hunk 10 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/sort/SliyaniemSort.java]: I've changed

File: src/main/java/mate/academy/hw02/sort/BuubleSort.java [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+                if (num[j] < num[j + 1]) {
+                    temp = num[j];
+                    num[j] = num[j + 1];
+                    num[j + 1] = temp;
+                    flag = true;
+                }
+
+            }
+            p++;
+            System.out.println("кол-во прохождений цикла Пузыр. сортировки = " + p);
```
Comment (Siril-O) [inline src/main/java/mate/academy/hw02/sort/BuubleSort.java:53]: No Russian, only English

**Review by Siril-O — APPROVED** [review]

File: src/main/java/mate/academy/hw02/sort/BuubleSort.java (hunk 8 above)
Comment (MateAcademy) [reply src/main/java/mate/academy/hw02/sort/BuubleSort.java]: I've changed

**Pull request merged**